### PR TITLE
[1.20.x] Early display fixes/workarounds for buggy drivers

### PR DESCRIPTION
**Pull request by PaintNinja**
This PR does some stuff that might help broken drivers detect Forge properly. Testing welcome.

- Comment out broken icon code
    - It always fails to set the icon because it doesn't exist, so don't bother running it. Best case it helps some drivers, worst case we skip wasting cycles for something we know doesn't work.
- Use the Vanilla window title initially
    - A window hint string is set for the title, but a completely different actual window title was set.
    - Now sets the window title to "Minecraft* 1.20.4" when creating the window, which might help
- Move the asterisk on the final window title
    - Forge used to set the title to "Minecraft Forge* 1.20.4", but this might be tripping up drivers that rely on the window title starting with "Minecraft*" to know if it's a modded environment
- Show a one-time help log immediately before the first attempt
    - Helps users that look at their debug.log/latest.log and just see `[main/INFO] [EARLYDISPLAY/]: Trying GL version 4.6` at the end with seemingly no error
    - Automatically turns itself off once a successful launch happens. Can be manually turned back on again in fml.toml (handy for swapping graphics cards for example)

### CODE REVIEW

**Comment by PaintNinja** [comment]
Installer to aid testing: 
[forge-1.20.4-49.0.46-1.20.x-early-display-fixes-installer.jar.zip](https://github.com/MinecraftForge/MinecraftForge/files/14968654/forge-1.20.4-49.0.46-1.20.x-early-display-fixes-installer.jar.zip)

File: fmlearlydisplay/src/main/java/net/minecraftforge/fml/earlydisplay/DisplayWindow.java [hunk 1]
```diff
@@ -402,11 +400,23 @@ public void initWindow(@Nullable String mcVersion) {
                 continue;
             }
             LOGGER.info("Trying GL version " + glVersionToTry);
+            if (showHelpLog && versidx == 0) {
+                LOGGER.info("""
+                If this message is the only thing at the bottom of your log before a crash, you probably have a driver issue.
+                
+                Possible solutions:
+                A) Make sure Minecraft is set to prefer high performance graphics in the OS and/or driver control panel
```
Comment (Jonathing) [inline fmlearlydisplay/src/main/java/net/minecraftforge/fml/earlydisplay/DisplayWindow.java:408]: I have had weird niche issues where simply setting Minecraft itself does not affect what graphics card Windows will choose to run the game with. However that is more of an issue with third-party launchers. The default Minecraft Launcher from the Microsoft Store will always request that Windows uses the high performance graphics card.

**Review by Jonathing — APPROVED** [review]
Cool. This is a relatively harmless PR. Unfortunately I don't have a computer shitty enough to test the defective hardware/driver cases, but the code looks good and the game starts up in production (vanilla Minecraft Launcher) without issues.

**General Requirements**
- [X] Game loads.

**PR-specific Requests**
- [ ] Optionally, clarify that the user might need to specify that their Java installation might also need to be set to use their dGPU. [See here.](https://github.com/MinecraftForge/MinecraftForge/pull/9921/files#r1564834067)

**Comment by Jonathing** [comment]
Also, I'm just going to trust that the development environment works well. I just tested the production environment.

File: fmlearlydisplay/src/main/java/net/minecraftforge/fml/earlydisplay/DisplayWindow.java (hunk 1 above)
Comment (PaintNinja) [reply fmlearlydisplay/src/main/java/net/minecraftforge/fml/earlydisplay/DisplayWindow.java:408]: I've found that the MS Store version sadly does not always request the high perf GPU, so recommending the official launcher isn't as robust of a solution as telling users to manually set the preference in their OS. :/

Comment (Jonathing) [reply fmlearlydisplay/src/main/java/net/minecraftforge/fml/earlydisplay/DisplayWindow.java:408]: I wasn't suggesting recommending the official launcher, I was suggesting adding that they might want to add the javaw.exe binary to their list of dGPU applications. The part about the official launcher using the dGPU is just commentary of my own experience.

Comment (PaintNinja) [reply fmlearlydisplay/src/main/java/net/minecraftforge/fml/earlydisplay/DisplayWindow.java:408]: Oh, I see. I deliberately left it as just "Make sure Minecraft is set..." to be clear that we don't really care how, just that it's working correctly. Can get a bit wordy if I change it to "Make sure anything Minecraft-related (such as launchers, Java, and so on...) are set..." and lead to some PC noobs setting their own Java install but not the one actually bundled with the launcher. If you have any suggestions for alternate wording, feel free

**Review by LexManos — APPROVED** [review]
I don't have any device with broken drivers to test the early screen issue. but everything loads up fine for me so lets hope this helps those with broken setups.
